### PR TITLE
Allow tags to be added when lease table is created

### PR DESCRIPTION
**Pull request by RyanFrench**
*Issue #, if available:* #154

*Description of changes:* Add capability to add tags when creating the lease table

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresherTest.java [hunk 1]
```diff
@@ -313,6 +317,40 @@ public void testCreateLeaseTableProvisionedBillingModeIfNotExists() throws Excep
         Assert.assertTrue(result);
     }
 
+    @Test
+    public void testCreateLeaseTableWithTagsIfNotExists() throws Exception {
+        tags = Arrays.asList(Tag.builder().key("foo").value("bar").build());
```
Comment (stair-aws) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresherTest.java:322]: IDE warning: `s/Arrays.asList/Collections.singletonList/`

**Review by stair-aws — CHANGES_REQUESTED** [review]
Thanks for the contribution! PR mostly LGTM, and verified `mvn clean test` against a rebase. A few small requested changes, and question.

**Comment by RyanFrench** [comment]
Thanks for taking a look and the guidance! I'm not a java dev by any means, so appreciate there's quite a bit in there I didn't know.

I believe I've addressed all of the changes requested and added my reasoning about using DSACL above.

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresherTest.java [hunk 2]
```diff
@@ -313,6 +317,40 @@ public void testCreateLeaseTableProvisionedBillingModeIfNotExists() throws Excep
...
+                .writeCapacityUnits(10L).build();
+        final CreateTableRequest createTableRequest = CreateTableRequest.builder()
+                .tableName(TABLE_NAME)
+                .keySchema(leaseSerializer.getKeySchema())
+                .attributeDefinitions(leaseSerializer.getAttributeDefinitions())
+                .provisionedThroughput(throughput)
+                .tags(tags)
+                .build();
+        when(dynamoDbClient.createTable(createTableRequest)).thenReturn(mockCreateTableFuture);
+        when(mockCreateTableFuture.get(eq(LeaseManagementConfig.DEFAULT_REQUEST_TIMEOUT.toMillis()), eq(TimeUnit.MILLISECONDS)))
```
[the remark this answers is not in the document]
Comment (stair-aws) [reply amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresherTest.java]: Strange. "Works on my machine" w/o this line (and I'm sync'd to `6783e1e`+rebase). It's an insignificant concern, so let's leave as-is for this PR.

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java [hunk 3]
```diff
@@ -787,6 +802,16 @@ protected DependencyException convertAndRethrowExceptions(String operation, Stri
         }
     }
 
+    private CreateTableRequest.Builder createTableRequestBuilder() {
+    final CreateTableRequest.Builder builder = CreateTableRequest.builder().tableName(table).keySchema(serializer.getKeySchema())
```
Comment (stair-aws) [inline amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java:806]: The indentation for code in this method is incorrect. (Looks like most lines should be indented +1 level?)

Would you kindly fix so the indentation matches the convention of other methods?

(FWIW, I normally regard whitespace as a nitpick. However, seeing `}` at position 1 is jarring when it isn't end-of-class.)

**Review by stair-aws — CHANGES_REQUESTED** [review]
LGTM (w/ one small request)!

To double-check: did you execute this in an AWS account and validate the DDB tags appear and will satisfy your ask in https://github.com/awslabs/amazon-kinesis-client/issues/154 ?

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java (hunk 3 above)
Comment (RyanFrench) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java]: Fixed!

**Comment by RyanFrench** [comment]
I did execute this in an AWS account and could see tags being added to the table correctly. I also tested to make sure that creating without tags still worked. I can add some integration tests for this the same as the others in the project, however, I seem to be hitting intermittent failures when running them from `PeriodicShardSyncManagerTest.testFor1000DifferentValidReshardHierarchyTreeWithSomeInProgressParentsTheHashRangesAreAlwaysComplete`

**Review by stair-aws — CHANGES_REQUESTED** [review]
@RyanFrench Would you kindly apply the following patch to your branch:
```
diff --git a/amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java b/amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java
index b06561a..8b60f6d 100644
--- a/amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java
+++ b/amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java
@@ -648,7 +648,7 @@ public class DynamoDBLeaseManagementFactory implements LeaseManagementFactory {
     @Override
     public DynamoDBLeaseRefresher createLeaseRefresher() {
         return new DynamoDBLeaseRefresher(tableName, dynamoDBClient, leaseSerializer, consistentReads,
-                tableCreatorCallback, dynamoDbRequestTimeout, billingMode);
+                tableCreatorCallback, dynamoDbRequestTimeout, billingMode, tags);
     }
 
     @Override
```
With this patch, I verified that a virgin KCL application created the DDB lease table w/ the provided tags.

Nice work!

**Review by stair-aws — APPROVED** [review]

**Pull request merged**